service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -189,18 +193,32 @@ func recordActivityTaskStarted(
 	}
 
 	if !pollerDeployment.Equal(wfDeployment) {
-		// Task is redirected, see if a transition can start.
-		if err := mutableState.StartDeploymentTransition(pollerDeployment); err != nil {
-			if errors.Is(err, workflow.ErrPinnedWorkflowCannotTransition) {
-				// This must be a task from a time that the workflow was unpinned, but it's
-				// now pinned so can't transition. Matching can drop the task safely.
-				return nil, false, serviceerrors.NewObsoleteMatchingTask(err.Error())
-			}
+		// Task is redirected, see if this activity should start a transition on the workflow. The
+		// workflow transition happens only if the workflow TQ's current deployment is the same as
+		// the poller deployment. Otherwise, it means the activity is independently versioned, we
```

Comment:
Also, the idea that we're switching behavior based on whether these match seems wrong, shouldn't we look at the configured versioning intention for the activity? Do we still have that concept in v3?

I see maybe the next PR addresses this?